### ISSUE TITLE
Fix report flags not reporting on subsequent runs when cache is used

### ISSUE THREAD

**Issue by XhmikosR** [stylelint/stylelint#5049]
> Clearly describe the bug

When using the `--rd` switch, if there is a needless disable, the second or any subsequent run doesn't report it.

> Which rule, if any, is the bug related to?

None

> What stylelint configuration is needed to reproduce the bug?

Probably doesn't matter:

```json
{
  "extends": [
    "stylelint-config-twbs-bootstrap/scss"
  ],
  "rules": {
    "function-disallowed-list": [
      "calc",
      "lighten",
      "darken"
    ],
    "property-disallowed-list": [
      "border-radius",
      "border-top-left-radius",
      "border-top-right-radius",
      "border-bottom-right-radius",
      "border-bottom-left-radius",
      "transition"
    ],
    "scss/dollar-variable-default": [
      true,
      {
        "ignore": "local"
      }
    ],
    "scss/selector-no-union-class-name": true
  }
}
```

> Which version of stylelint are you using?

13.7.2 and 13.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `stylelint "**/*.{css,scss}" --cache --cache-location .cache/.stylelintcache --rd`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Probably not

> What did you expect to happen?

Any needless disables should be reported on subsequent runs.

> What actually happened (e.g. what warnings or errors did you get)?

No needless disables were reported. **If I delete the cache or remove the `--cache` CLI flag, then the needless disables are reported**.

@jeddy3 I verified this and indeed happens with 13.7.0, 13.7.2 and 13.8.0 that I could quickly test.

**Comment by jeddy3** [comment]
> @jeddy3 I verified this and indeed happens with both 13.7.2 and 13.8.0.

Thanks! It looks to be a longer-standing bug than a regression with 13.8.0.

I've labelled the issue as ready to implement. If anyone has time to look into this then please consider [contributing](https://stylelint.io/contributing).

**Comment by XhmikosR** [comment]
I tested 13.7.0 and also has the issue. I can't go further behind because of the rule's renaming (well, I can but time 🙂)

**Comment by ybiquitous** [comment]
I'm revisiting this issue. Perhaps, here's a reproduction with 16.2.0:

`stylelint.config.mjs`:

```js
export default {
	rules: {
		'block-no-empty': true,
	},
};
```

`test.css`:

```css
a { color: red; } /* stylelint-disable-line block-no-empty */
```

```sh-session
$ npx stylelint -v
16.2.0

$ rm -f .stylelintcache

$ DEBUG=* npx stylelint "*.css" --rd --cache
  stylelint:file-cache Cache file is created at /Users/masafumi.koba/git/stylelint/stylelint/tmp/.stylelintcache +0ms
  stylelint:standalone Processing /Users/masafumi.koba/git/stylelint/stylelint/tmp/test.css +0ms
  stylelint:standalone Linting complete in 69ms +16ms

test.css
 1:19  ✖  Needless disable for "block-no-empty"  --report-needless-disables

1 problem (1 error, 0 warnings)

$ DEBUG=* npx stylelint "*.css" --rd --cache
  stylelint:file-cache Cache file is created at /Users/masafumi.koba/git/stylelint/stylelint/tmp/.stylelintcache +0ms
  stylelint:standalone Processing /Users/masafumi.koba/git/stylelint/stylelint/tmp/test.css +0ms
  stylelint:file-cache Skip linting /Users/masafumi.koba/git/stylelint/stylelint/tmp/test.css. File hasn't changed. +33ms
  stylelint:standalone Linting complete in 59ms +7ms
```

---

Deleting cache doesn't seem to work when `report-needless-disables` errors happen: 🤔 

https://github.com/stylelint/stylelint/blob/eebb78694ee87e76245414108b0ae2626c78729f/lib/standalone.mjs#L219-L225

**Comment by ybiquitous** [comment]
I found a problem in the code. I'll open a pull request soon.
